core/trino-main/src/main/java/io/trino/operator/DynamicFilterSourceOperator.java

```diff
@@ -370,42 +242,16 @@ public void finish()
             return;
         }
         finished = true;
-        ImmutableMap.Builder<DynamicFilterId, Domain> domainsBuilder = ImmutableMap.builder();
-        if (valueSets == null) {
-            if (minValues == null) {
-                // there were too many rows to collect min/max range
-                // dynamicPredicateConsumer was notified with 'all' in handleTooLargePredicate if there are no orderable types,
-                // else it was notified with 'all' in handleMinMaxCollectionLimitExceeded
-                return;
-            }
-            // valueSets became too large, create TupleDomain from min/max values
-            for (int i = 0; i < minMaxChannels.size(); i++) {
-                Integer channelIndex = minMaxChannels.get(i);
-                Type type = channels.get(channelIndex).type;
-                if (minValues[i] == null) {
-                    // all values were null
-                    domainsBuilder.put(channels.get(channelIndex).filterId, Domain.none(type));
-                    continue;
-                }
-                Object min = blockToNativeValue(type, minValues[i]);
-                Object max = blockToNativeValue(type, maxValues[i]);
-                Domain domain = Domain.create(
-                        ValueSet.ofRanges(range(type, min, true, max, true)),
-                        false);
-                domainsBuilder.put(channels.get(channelIndex).filterId, domain);
-            }
-            minValues = null;
-            maxValues = null;
-            dynamicPredicateConsumer.addPartition(TupleDomain.withColumnDomains(domainsBuilder.buildOrThrow()));
+        if (emptyDomain) {
+            // dynamicPredicateConsumer was already notified with 'all'
             return;
         }
+
+        ImmutableMap.Builder<DynamicFilterId, Domain> domainsBuilder = ImmutableMap.builder();
         for (int channelIndex = 0; channelIndex < channels.size(); ++channelIndex) {
-            Block block = blockBuilders[channelIndex].build();
-            Type type = channels.get(channelIndex).type;
-            domainsBuilder.put(channels.get(channelIndex).filterId, convertToDomain(type, block));
+            DynamicFilterId filterId = channels.get(channelIndex).filterId;
+            channelFilters[channelIndex].getDomain().ifPresent(domain -> domainsBuilder.put(filterId, domain));
         }
-        valueSets = null;
-        blockBuilders = null;
         dynamicPredicateConsumer.addPartition(TupleDomain.withColumnDomains(domainsBuilder.buildOrThrow()));
     }
 
```

Comment:
`convertToDomain` should now be a private static method or a private method of `ChannelFilter`